### PR TITLE
Reduce amount of success messages sent to users

### LINKED ISSUE #714: Snackbar fires off too many "data loaded" messages
In Imviz, when one data file is loaded, you see two different snackbar messages that tell you the same thing.

1. A generic "data loaded" message.
2. A slightly more specific "`<data label>` is loaded" message.

This might even be worse in a case where a lot of data is loaded at once, e.g., in Mosviz.

We need to revisit the need to fire off so many similar messages and update the logic in affected parses, as needed.

[🐱](https://jira.stsci.edu/browse/JDAT-1587)

p.s. This was exposed after snackbar message queue was fixed in #701

### PR DESCRIPTION
**Pull request by rosteen**
### Description

Don't overwhelm users with a million success messages (e.g. for every 1D/2D spectrum and image in Mosviz). @pllim I tried not to touch the image parser in Mosviz for now pending #783  😁.

Fixes #714 .

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by rosteen** [comment]
Note that a large part of this is homogenizing the parsers to all use `app.add_data` rather than adding things to `app.data_collection` directly, which bypassed the success message sent by `add_data`.

**Comment by pllim** [comment]
So, this supersedes #720 ?

File: jdaviz/configs/mosviz/plugins/parsers.py [hunk 1]
```diff
@@ -644,13 +644,10 @@ def mos_niriss_parser(app, data_dir, obs_label=""):
     with app.data_collection.delay_link_manager_update():
 
         for label, data in add_to_glue.items():
-            app.data_collection[label] = data
+            app.add_data(data, label, notify_done=False)
 
         # We then populate the table inside this context manager as _add_to_table
         # does operations that also trigger link manager updates.
-
-        print("Populating table")
```
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/parsers.py:652]: I thought @javerbukh wanted to keep these `print` functions when I suggested to remove them in a different PR? 🤷

Comment (rosteen) [reply jdaviz/configs/mosviz/plugins/parsers.py:652]: Oh, I wasn't aware of that conversation, I thought they were prints for debugging that snuck into main.

Comment (rosteen) [reply jdaviz/configs/mosviz/plugins/parsers.py:654]: @javerbukh confirmed in our dev meeting that these were relics from when loading took a very long time and are irrelevant now. He's ok with removing them.

**Review by javerbukh — APPROVED** [review]
Looks good! Just a thought, if we're decreasing the amount of "data loaded messages", should we turn them from gray to green? I got some small dopamine from seeing that in mosviz and missed it a little bit when seeing the gray load data message in cubeviz and specviz. Feel free to ignore!

**Comment by rosteen** [comment]
Good call! I think I had meant to do that and it slipped my mind.

**Comment by pllim** [comment]
Ops... conflict 😬

**Comment by rosteen** [comment]
Not anymore, although apparently rebasing changed the code coverage.

**Review by ibusko — APPROVED** [review]
Works good. Nice job in removing the annoying flurry of snackbar messages!

**Review by pllim — APPROVED** [review]
Definitely much better in Imviz now. ⭐⭐⭐⭐⭐

**Pull request merged**